From 5c935af7d3b3a3c9feb4e0d9ea214b6d25a7a2e7 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Tue, 19 Jul 2022 18:31:04 +0200
Subject: [PATCH] build: Port to gcr4

The gcr4 is going to replace gcr3. As only base functions are used,
the port to gcr4 is trivial.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2386>
---
 js/ui/environment.js           | 2 +-
 meson.build                    | 4 ++--
 src/meson.build                | 2 +-
 src/shell-keyring-prompt.c     | 6 +++---
 src/shell-secure-text-buffer.c | 2 +-
 5 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/js/ui/environment.js b/js/ui/environment.js
index 83df7bcce4..4853a83d55 100644
--- a/js/ui/environment.js
+++ b/js/ui/environment.js
@@ -8,7 +8,7 @@ imports.gi.versions.Atk = '1.0';
 imports.gi.versions.Atspi = '2.0';
 imports.gi.versions.Clutter = Config.LIBMUTTER_API_VERSION;
 imports.gi.versions.Cogl = Config.LIBMUTTER_API_VERSION;
-imports.gi.versions.Gcr = '3';
+imports.gi.versions.Gcr = '4';
 imports.gi.versions.Gdk = '3.0';
 imports.gi.versions.Gdm = '1.0';
 imports.gi.versions.Geoclue = '2.0';
diff --git a/meson.build b/meson.build
index ea13eba17e..a7a24c61ab 100644
--- a/meson.build
+++ b/meson.build
@@ -20,7 +20,7 @@ libmutter_test_pc = 'libmutter-test-' + mutter_api_version
 
 ecal_req = '>= 3.33.1'
 eds_req = '>= 3.33.1'
-gcr_req = '>= 3.7.5'
+gcr_req = '>= 3.90.0'
 gio_req = '>= 2.56.0'
 gi_req = '>= 1.49.1'
 gjs_req = '>= 1.73.1'
@@ -71,7 +71,7 @@ endif
 atk_bridge_dep = dependency('atk-bridge-2.0')
 ecal_dep = dependency('libecal-2.0', version: ecal_req)
 eds_dep = dependency('libedataserver-1.2', version: eds_req)
-gcr_dep = dependency('gcr-base-3', version: gcr_req)
+gcr_dep = dependency('gcr-4', version: gcr_req)
 gdk_x11_dep = dependency('gdk-x11-3.0')
 gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0')
 gi_dep = dependency('gobject-introspection-1.0', version: gi_req)
diff --git a/src/meson.build b/src/meson.build
index 112782d862..e784a5132f 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -213,7 +213,7 @@ libshell_dep = declare_dependency(link_with: libshell)
 libshell_gir_includes = [
   'Clutter-@0@'.format(mutter_api_version),
   'Meta-@0@'.format(mutter_api_version),
-  'Gcr-3',
+  'Gcr-4',
   'PolkitAgent-1.0'
 ]
 
diff --git a/src/shell-keyring-prompt.c b/src/shell-keyring-prompt.c
index 83c6746389..bb03279666 100644
--- a/src/shell-keyring-prompt.c
+++ b/src/shell-keyring-prompt.c
@@ -26,7 +26,7 @@
 #include "shell-secure-text-buffer.h"
 
 #define GCR_API_SUBJECT_TO_CHANGE
-#include <gcr/gcr-base.h>
+#include <gcr/gcr.h>
 
 #include <glib/gi18n.h>
 
@@ -91,7 +91,7 @@ enum {
 
 static GParamSpec *props[N_PROPS] = { NULL, };
 
-static void    shell_keyring_prompt_iface     (GcrPromptIface *iface);
+static void    shell_keyring_prompt_iface     (GcrPromptInterface *iface);
 
 G_DEFINE_TYPE_WITH_CODE (ShellKeyringPrompt, shell_keyring_prompt, G_TYPE_OBJECT,
                          G_IMPLEMENT_INTERFACE (GCR_TYPE_PROMPT, shell_keyring_prompt_iface);
@@ -531,7 +531,7 @@ shell_keyring_prompt_close (GcrPrompt *prompt)
 }
 
 static void
-shell_keyring_prompt_iface (GcrPromptIface *iface)
+shell_keyring_prompt_iface (GcrPromptInterface *iface)
 {
   iface->prompt_password_async = shell_keyring_prompt_password_async;
   iface->prompt_password_finish = shell_keyring_prompt_password_finish;
diff --git a/src/shell-secure-text-buffer.c b/src/shell-secure-text-buffer.c
index 03af451ba9..8271410e22 100644
--- a/src/shell-secure-text-buffer.c
+++ b/src/shell-secure-text-buffer.c
@@ -26,7 +26,7 @@
 #include "shell-secure-text-buffer.h"
 
 #define GCR_API_SUBJECT_TO_CHANGE
-#include <gcr/gcr-base.h>
+#include <gcr/gcr.h>
 
 #include <string.h>
 
-- 
GitLab

